tools/msgpack4s/src/main/scala/org.velvia/msgpack/PlayJsonCodecs.scala

```diff
@@ -28,16 +26,33 @@ object PlayJsonCodecs {
   }
 
   implicit object JsNumberCodec extends Codec[JsNumber] {
-    def pack(out: DataOutputStream, item: JsNumber): Unit = { BigDecimalCodec.pack(out, item.value.underlying) }
-    val unpackFuncMap = BigDecimalCodec.unpackFuncMap.mapValues(_.andThen(b ⇒ JsNumber(BigDecimal(b))))
+    def pack(out: DataOutputStream, item: JsNumber): Unit = {
+      val value: BigDecimal = item.value
+
+      // TODO: Implement special case here, something with precision and scale modification
+      if (value.isValidLong)
+        LongCodec.pack(out, value.longValue())
+      else if (value.isExactFloat) // TODO: Check this method, it does not return True even for value 2.7
+        FloatCodec.pack(out, value.floatValue())
+      else if (value.isExactDouble)
+        DoubleCodec.pack(out, value.doubleValue())
+      else
+        DoubleCodec.pack(out, value.underlying().doubleValue())
+    }
+
```

Comment:
@vananyev I'm not certain, however could you please check the following way
```
def pack(out: DataOutputStream, item: JsNumber) = item.value match {
    case value if value.isValidLong => LongCodec.pack(out, value.longValue())
    case value if value.isExactFloat => FloatCodec.pack(out, value.floatValue())
    case value if value.isExactDouble => DoubleCodec.pack(out, value.doubleValue())
    case value => DoubleCodec.pack(out, value.underlying().doubleValue())
  }
```

Reply:
I would say 'if-else' statement more descriptive in that case.